File: task-scheduler/go/sprint.go
package main

import "fmt"

type Sprint struct{
  id int
  name string
  startDate int
  endDate int
  tasks []Task
}


func (s *Sprint) addTask(task Task)bool{
  s.tasks = append(s.tasks, task)
  return true
}
func (s *Sprint) removeTask(task Task) bool{
  for i, v := range s.tasks{
    if v.id == task.id{
      //remove element 
      s.tasks = append(s.tasks[:i], s.tasks[i+1:]...)
      return true
    }
  }
  return false
}

func (s *Sprint) printDelayTasks(){
  for _, v := range s.tasks{
    if v.status != DONE{
      fmt.Println("delay task:", v)
    }
  }
}

File: task-scheduler/typescript/sprint.ts
import { Task, TicketStatus } from "./ticket";

export class Sprint {
  id: number;
  name: string;
  startDate: number;
  endDate: number;
  tickets: Task[]


  constructor(name: string, endDate: number, startDate: number, tasks: Task[]){
    this.id = Math.floor(Math.random() * 11);
    this.name = name;
    this.startDate = startDate;
    this.endDate = endDate;
    this.tickets = tasks
  }
  public addTask(task: Task){
    this.tickets.push(task)
  }

  public printDelayedTickets(){
    for (const task of this.tickets){
      if (task.ticketStatus !== TicketStatus.CLOSED){
        console.log(task)
      }
    }
  }
}



